lib/perl/Genome/Ptero/Wrapper.pm

```diff
@@ -1,12 +1,198 @@
 package Genome::Ptero::Wrapper;
 
 use strict;
-use warnings;
+use warnings FATAL => qw(all);
 use Genome;
+use Data::Dump qw();
+use IO::Handle;
+use Cwd qw(abs_path);
+use Ptero::Proxy::Workflow::Execution;
 
 class Genome::Ptero::Wrapper {
-    is => 'Command::Tree',
-    doc => 'execution wrappers for the Ptero system',
+    is => 'Command::V2',
+    has => [
+        'command_class' => {
+            is => 'Text',
+            doc => 'class name of the Genome Command to execute',
+        },
+        'method' => {
+            is => 'Text',
+            valid_values => ['execute', 'shortcut'],
+            doc => 'method to call on the Genome Command object',
+        },
+        'log_directory' => {
+            is => 'Path',
+            doc => "Where the log files should be stored (Their names will " .
+                "be automatically generated.)",
+        },
+    ],
+    has_transient_optional => [
+        'execution' => {
+            is => 'Ptero::Proxy::Workflow::Execution',
+        }
+    ],
+
+    doc => 'Ptero execution wrapper for Genome Command objects',
 };
 
+
+sub execute {
+    my $self = shift;
+
+    Genome::Sys->create_directory($self->log_directory);
+    validate_environment();
+
+    $self->execution(Ptero::Proxy::Workflow::Execution->new(
+        $ENV{PTERO_WORKFLOW_EXECUTION_URL}));
+
+    # this will get saved to the execution's stderr/stdout
+    $self->_log_execution_information;
+
+    $self->_setup_logging;
+
+    # this will get logged to the log files
+    $self->_log_execution_information;
+
+    my $command = $self->_instantiate_command($self->execution->inputs);
+
+    $self->_run_command($command);
+
+    $self->_teardown_logging;
+    $self->execution->set_outputs(
+        _get_command_outputs($command, $self->command_class));
+
+    return 1;
+}
+
+sub validate_environment {
+    my $self = shift;
+    unless (defined $ENV{PTERO_WORKFLOW_EXECUTION_URL}) {
+        die $self->error_message("Environment variable PTERO_WORKFLOW_EXECUTION_URL must be set");
+        exit 1;
+    }
+}
+
+sub _stdout_log_path {
+    my $self = shift;
+
+    my $base_name = $self->execution->name;
+    my $output_log = File::Spec->join(abs_path($self->log_directory), "$base_name.out");
```

Comment:
They are guaranteed unique now, execution name is unique, as for the length, I'll see if we can handle that in execution name.